### PR TITLE
mandatory transactions are stuck to round

### LINKED ISSUE #2398: should validate amount of service transactions
For pay-fees and generate-challenge transactions check that every block contains exact number of transactions required

### PR DESCRIPTION
**Pull request by dabasov**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/block_reward.go [hunk 1]
```diff
@@ -35,32 +39,41 @@ func (ssc *StorageSmartContract) blobberBlockRewards(
 
 	conf, err := ssc.getConfig(balances, true)
 	if err != nil {
-		return common.NewError("blobber_block_rewards_failed",
+		return "", common.NewError("blobber_block_rewards_failed",
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/block_reward.go]: Seems we are returning empty string in all cases, perhaps we don't need it or return something as transaction output?

**Comment by dabasov** [comment]
fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Pull request merged**